NEXT-23243 - Add RangeAggregation (#785)
shopware/docs: resources/references/core-reference/dal-reference/aggregations-reference.md

{% tab title="API Criteria" %}
Request

> ```javascript

Reviewer: @trandangtri on the top of the file there is a table of all existing aggregations, please add the new aggregation as well there

Author: Thanks @keulinho . Just updated it.

{% endtab %}
{% endtabs %}

## Range aggregations

> Allow to aggregate data on predefined ranges values for more flexibility in the DAL.
> It could allow to provide faceted filters by predefined range for example.
> 
> Bound are computed in SQL as in the elasticsearch native range aggregation:
> * `from` will be compared with greater than or equal
> * `to` will be compared with lower than

Reviewer 2: ```suggestion
Allows to aggregate data on predefined range of values for more flexibility in the DAL - for example, it provides faceted filters on predefined range.

Bound are computed in SQL as in the elasticsearch native range aggregation:
* `from` will be compared with greater than or equal to
* `to` will be compared with lower than
```
